### ISSUE TITLE
[BUG] wacom pen stops reacting soon after getting in Retopoflow workspace

### ISSUE THREAD

**Issue by CROiga** [CGCookie/retopoflow#1188]
Retopoflow Version: v 3.2.9

Blender Version/Hash: 3.3.3 , 3.4 , 3.5
Platform Version/Distribution: Windows 11

**Issue:**

Wacom pen on my Wacom Cintiq stops reacting very soon after getting into RetopoFlow workspace... it works for a while, and then just stops reacting,.... I need to connect my mouse to get out of retopoflow workspace, and then the pen starts working again. Touch screen of my cintiq works just fine. I have to say that Im a total noob, since this was the first time I installed the retopoflow plugin, and I dont know how to work with it,.. but than again, this seems like an issue to me, so I thought it would be a good idea to report back.
Tested the issue on 3.3 3.4 and 3.5 versions of Blender,.. all react the same.

**Comment by vxlcoder** [comment]
thanks for the report, @CROiga.

I do not have a wacom cintiq on hand to test, but i do have a wacom intuous.  I just ran many tests, but haven't seen any issues yet.

Can I have you try screen capturing your editing session?  this would be especially useful if you can replicate this issue with factory settings.  Also, once the issue happens, try exiting RF, show the windows terminal in Blender (Window -> Toggle System Consol), copy _all_ the text in the window (might be a lot!), and paste the results here.  there might be some debugging info printed there that can help us figure out what the cause is.

thanks!

**Comment by jlampel** [comment]
I've been using an Intuos as well and have not seen this issue, but it could be specific to Cintiqs. I'll be on the lookout for any other reports of this.

**Comment by CROiga** [comment]
Sorry guys, have a kid with some problems so I couldn’t respond earlier. I will try to replicate the issue and record the screen while doing it. And get you the console text after that.

**Comment by vxlcoder** [comment]
completely understand. thanks!

**Comment by CROiga** [comment]
Ok, so here is the link to video recording. it is 85Mb so I uploaded it to wetransfer,... 

https://we.tl/t-iwDNomQHQb?utm_campaign=TRN_TDL_05&utm_source=sendgrid&utm_medium=email&trk=TRN_TDL_05

![Screenshot_20230215_125353](https://user-images.githubusercontent.com/8853273/219022185-94eb51f0-f478-4cc2-b221-c54992dbfacc.png)

and here is the content of a console window... not much at all, since the issue appeared in few minutes of work.

Read prefs: C:\Users\IGOR\AppData\Roaming\Blender Foundation\Blender\3.3\config\userpref.blend
RetopoFlow: Initial load
Addon Common: (shaders) GLSL Version: 4.50 NVIDIA
Addon Common: compiling UI shader
Addon Common: batching for shader
Addon Common: UI shader initialized
Addon Common: ########################################
              Loading shader using Shader
RetopoFlow Shader Info: uiShader (22)
  attribs:
  uniforms: uMVPMatrix (23), left (12), right (21), top (22), bottom (7), width (25), height (9), depth (8), margin_left (14), margin_right (15), margin_top (16), margin_bottom (13), padding_left (18), padding_right (19), padding_top (20), padding_bottom (17), border_width (6), border_radius (3), border_left_color (2), border_right_color (4), border_top_color (5), border_bottom_color (1), background_color (0), using_image (24), image_fit (11), image (10)
RetopoFlow git: None
gpustate.gpu_info cannot get shader version!
RetopoFlow options path: C:\Users\IGOR\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\RetopoFlow\RetopoFlow_options.json
Addon Common: (bmesh_render) GLSL Version: NVIDIA Corporation, NVIDIA GeForce RTX 4090/PCIe/SSE2, 4.5.0 NVIDIA 528.24, (no shader version)
cgcookie addon: Verbose is enabled
cgcookie addon: Read in JSON settings from file
RetopoFlow keymaps path: C:\Users\IGOR\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\RetopoFlow\RetopoFlow_keymaps.json
RetopoFlow: setting up scaling factor
  Unit scaling factor: 0.16590082606999204
RetopoFlow: scaling view, sources, and target by 6.03
RetopoFlow: storing clip distances: 0.009999999776482582 1000.0
Addon Common Warning: Cannot toggle header visibility (addon_common/common/blender.py: toggle_screen_header)
  Skipping while bug exists in Blender 3.0+, see: https://developer.blender.org/T93410
Addon Common: Loaded font "C:\Users\IGOR\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\RetopoFlow\addon_common\common\fonts\DroidSans-Blender.ttf" as id 3
RetopoFlow: Pausing help image preloading (1676461788.2870593)
RetopoFlow: Setting up target mesh (0.006000041961669922)
RetopoFlow: Setting up source mesh(es) (0.003999471664428711)
  rfsources...
  bboxes...
  drawing...
  done!
RetopoFlow: Setting up symmetry data structures (0.0029997825622558594)
RetopoFlow: Setting up rotation target (0.003000497817993164)
RetopoFlow: Setting up RetopoFlow states (0.005000114440917969)
RetopoFlow: Setting up RetopoFlow tools (0.003000974655151367)
RetopoFlow: Setting up grease marks (0.002001047134399414)
RetopoFlow: Setting up visualizations (0.007001161575317383)
RetopoFlow: Setting up user interface (0.007001161575317383)
RetopoFlow: Setting up undo system (0.2660946846008301)
RetopoFlow: Checking auto save / save (0.022998571395874023)
Addon Common: Loaded font "C:\Users\IGOR\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\RetopoFlow\addon_common\common\fonts\OpenSans-Italic.ttf" as id 4
Addon Common: Loaded font "C:\Users\IGOR\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\RetopoFlow\addon_common\common\fonts\DejaVuSansMono.ttf" as id 5
RetopoFlow: Checking target symmetry (0.022000789642333984)
RetopoFlow: Loading welcome message (0.007999897003173828)
Addon Common: Loaded font "C:\Users\IGOR\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\RetopoFlow\addon_common\common\fonts\OpenSans-Bold.ttf" as id 6
RetopoFlow: Resuming help image preloading (0.25333166122436523)
RetopoFlow: done with start
RetopoFlow: scaling view, sources, and target by 0.17

**Comment by CROiga** [comment]
I repeated the test a few times, and the problem appeared using different tools, so it is not tool related,... also,.. it was a clean version of Blender 3.3.3 with no other add-ons enabled... I clean installed it and used a clean new config, nothing is changed.

**Comment by vxlcoder** [comment]
hi @CROiga.  Sorry for my delayed reply.  Let's try this...

Download and open this [.blend](https://www.dropbox.com/s/yp2qm354t6il7ba/modal_event_reporting.blend?dl=0).  You'll see some code in the bottom left corner and a blank text in bottom right.  Click the Run Script button in the header of script (see screenshot below).  This will start recording all the events that occur for the next 10 seconds into the blank text on right.  You can see in the status bar how much time has elapsed since the script started.  All actions should pass through to Blender as normal.  So aside from the status bar and maybe the text on right updating, it _should_ look and feel like nothing unusual is going on.  You may run the script multiple times.  After you're done, save the .blend file and email to retopoflow@cgcookie.com along with details on what you recorded.  Also helpful might be to send your keymap (Edit > Preferences > Keymap > Export...) in the email.  These should help us see what you're doing that we are unable to replicate on our end.

TL;DR: Download .blend, run the script, use the tablet for the next 10 seconds, save and email us the .blend file.

![image](https://user-images.githubusercontent.com/3790699/233977742-cbc19919-d77a-4a1a-ade4-0a2b0f01711e.png)

Thanks!

**Comment by vxlcoder** [comment]
@CROiga, we've made some significant progress on handling inputs.

Can you try downloading and installing the version linked below to see if you continue to have issues?

https://www.dropbox.com/s/16xu7mowqkbm3mm/RetopoFlow_v3.4.0-alpha-GitHub.zip?dl=0

thanks for helping us debug this issue!

**Comment by jlampel** [comment]
Closing for lack of a response and lack of replication, but please let us know if you continue to have this issue!
